src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1593,8 +1593,13 @@ private byte[] initCEN() throws IOException {
             if (method != METHOD_STORED && method != METHOD_DEFLATED) {
                 throw new ZipException("invalid CEN header (unsupported compression method: " + method + ")");
             }
-            if (pos + CENHDR + nlen > limit) {
-                throw new ZipException("invalid CEN header (bad header size)");
+            long headerSize = (long)CENHDR + nlen + clen + elen;
```

Comment:
If the corresponding ZipFile local variable is renamed, this should also be updated.